db.ts

```diff
@@ -1,16 +1,20 @@
-const mongoose = require("mongoose");
+import { Response } from "express";
+import mongoose, { ConnectOptions } from "mongoose";
 
-const connectDB = async () => {
+export const connectDB = async (): Promise<void> => {
   try {
-    await mongoose.connect(process.env.DB_URL);
+    await mongoose.connect(process.env.DB_URL as string, {
```

Comment:
according to the logs, `useNewUrlParser` and `useUnifiedTopology` are deprecated and not needed. Remove them